<a href="https://colab.research.google.com/github/francomanca93/liberalismo-y-medioambiente/blob/master/Notebooks/1_Extraccion_y_limpieza_de_datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Extracción de datos. Conociendo a los mismos.

Este documento es sobre como se irán limpiando los datos para el proyecto Liberalismo y Medioambiente para la fundación Republica Liberal

Se irán agregando los enlaces a los datasets de los datos correspondientes.

[EPI](https://epi.yale.edu/downloads)

[Index of Economic Freedom - 2020](https://www.heritage.org/index/explore?u=637340655378618037)

[Index of Economic Freedom - 1995/2020](https://www.heritage.org/index/explore?view=by-region-country-year&u=637340656550516325)



## Configuramos Google Colab y librerias

El siguiente chunk de codigo sirve para ingresar a mi unidad virtual de Drive para trabajar con los datasets descargados de los enlaces anteriores.

In [1]:
from google.colab import drive
drive.mount('/content/drive/')

Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount("/content/drive/", force_remount=True).


In [2]:
ls '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/'

epi/  index_economic_freedom/


In [3]:
import pandas as pd
import numpy as np

## Conociendo Enviroment performance index

Leemos nuestro dataset, lo guardamos en una variable para luego reconocer las labels que tiene el dataframe.

In [4]:
ls '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/epi'

epi2020countryattributes20200604.csv  epi2020resultsregions20200604.csv
epi2020results20200604.csv            epi2020weights20200604.csv
epi2020resultsgroups20200604.csv


In [5]:
cd '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/epi'

/content/drive/My Drive/Colab Notebooks/republica-liberal/data/epi


Veamos que tiene el dataset **epi2020results...**

In [6]:
df_epi = pd.read_csv('epi2020results20200604.csv')
df_epi.head(5)

code  iso               country  ...  SNM.rnk.new  WRS.rnk.new  WWT.rnk.new
0     4  AFG           Afghanistan  ...           48          134          134
1    24  AGO                Angola  ...          123          134          134
2     8  ALB               Albania  ...           94           92           92
3   784  ARE  United Arab Emirates  ...          169           21           21
4    32  ARG             Argentina  ...            2           77           77

[5 rows x 141 columns]

Conozcamos la dimensión del EPI.

In [7]:
df_epi.shape

(180, 141)

Conozcamos el otro dataset, el mismo nos dice que tenemos las regiones, por el titulo, **epi2020resultsregions**

In [8]:
df_epi_region = pd.read_csv('epi2020resultsregions20200604.csv')
df_epi_region.head(5)

code  iso               country  ... WRS.rgn.rank  WWT.rgn.mean  WWT.rgn.rank
0     4  AFG           Afghanistan  ...            4      0.825000             4
1    24  AGO                Angola  ...           22      1.519565            22
2     8  ALB               Albania  ...           16     41.731579            16
3   784  ARE  United Arab Emirates  ...            3     36.700000             3
4    32  ARG             Argentina  ...           10      9.681250            10

[5 rows x 142 columns]

In [9]:
df_epi_region.shape

(180, 142)

Tiene 180 filas y 142 columnas. 

Obtengamos una lista de todas las columnas que tiene nuestro dataset. 

In [10]:
list(df_epi_region.columns)

['code',
 'iso',
 'country',
 'region',
 'EPI.new',
 'HLT.new',
 'AIR.new',
 'PMD.new',
 'HAD.new',
 'OZD.new',
 'H2O.new',
 'USD.new',
 'UWD.new',
 'HMT.new',
 'PBD.new',
 'WMG.new',
 'MSW.new',
 'ECO.new',
 'BDH.new',
 'TBN.new',
 'TBG.new',
 'MPA.new',
 'PAR.new',
 'SHI.new',
 'SPI.new',
 'BHV.new',
 'ECS.new',
 'TCL.new',
 'GRL.new',
 'WTL.new',
 'FSH.new',
 'FSS.new',
 'RMS.new',
 'FGT.new',
 'CCH.new',
 'CDA.new',
 'CHA.new',
 'FGA.new',
 'NDA.new',
 'BCA.new',
 'LCB.new',
 'GIB.new',
 'GHP.new',
 'APE.new',
 'SDA.new',
 'NXA.new',
 'AGR.new',
 'SNM.new',
 'WRS.new',
 'WWT.new',
 'EPI.rgn.mean',
 'EPI.rgn.rank',
 'HLT.rgn.mean',
 'HLT.rgn.rank',
 'AIR.rgn.mean',
 'AIR.rgn.rank',
 'PMD.rgn.mean',
 'PMD.rgn.rank',
 'HAD.rgn.mean',
 'HAD.rgn.rank',
 'OZD.rgn.mean',
 'OZD.rgn.rank',
 'H2O.rgn.mean',
 'H2O.rgn.rank',
 'USD.rgn.mean',
 'USD.rgn.rank',
 'UWD.rgn.mean',
 'UWD.rgn.rank',
 'HMT.rgn.mean',
 'HMT.rgn.rank',
 'PBD.rgn.mean',
 'PBD.rgn.rank',
 'WMG.rgn.mean',
 'WMG.rgn.rank',


Vemos que son muchas columnas. Vamos a buscar que significa cada abreviación.

En la siguiente tabla se puede ver mas información acerca de los indicadores anteriores. 

La tabla se encuentra en un pdf en [EPI Downloads](https://epi.yale.edu/downloads).

La misma fue obtenida del siguiente enlace:
[Technical Appendix](https://epi.yale.edu/downloads/epi2020technicalappendix20200803.pdf)

![Tabla EPI](https://imgur.com/fzFm1cZ.png)

El [índice de rendimiento ambiental o índice de desempeño ambiental](https://es.wikipedia.org/wiki/%C3%8Dndice_de_desempe%C3%B1o_ambiental) (Inglés: Environmental Performance Index, siglas EPI) es un método para cuantificar y clasificar numéricamente el desempeño ambiental de las políticas de un país. 

Se puede observar 2 grandes indicadores que contienen a un total de 11 y que estos ultimos contienen un total de 32. 

Los 2 indicadores que contienen a los demas son: 

-----------------------------

1. **El Environmental Health (HLT) tiene un 40% de peso en el Environment Performance Index (EPI).**

Este indicador nos cuantifica la salud ambiental de un pais. La salud ambiental nos dice los factores en el medio ambiente que pueden potencialmente afectar adversamente la salud de presentes y futuras generaciones.

Los indicadores que afectan a este son:
- La [calidad de aire](https://es.wikipedia.org/wiki/Calidad_del_aire) (Air Quality). Peso de 50%.
- El [saneamiento básico](https://es.wikipedia.org/wiki/Saneamiento_ambiental) y el [agua potable](https://es.wikipedia.org/wiki/Agua_potable) (Sanitation and Dringking water). Peso de 40%.
- Metales pesados (Heavy Metals). Peso de 5%. ([Lead Exposure](https://en.wikipedia.org/wiki/Lead_poisoning))
- [Gestion de residuos](https://es.wikipedia.org/wiki/Gesti%C3%B3n_de_residuos) (Waste Management). Peso de 5%.

---------------------

2. **El Ecosystem Vitality (ECO) tiene un 60% de peso en el Environment Performance Index (EPI).**

La vitalidad del [ecosistema](https://es.wikipedia.org/wiki/Ecosistema) se refiere a la actividad o energía para vivir o desarrollarse del mismo. Si tiene una baja vitalidad, es un indicador que el pais no es amigable con su ecosistema. 

Los indicadores que afectan a ese son:

- [Biodiversidad](https://es.wikipedia.org/wiki/Biodiversidad) y [habitat](https://es.wikipedia.org/wiki/H%C3%A1bitat) (Biodiversity & Habitat). Peso de 25%.
- [Servicios del ecosistema](https://es.wikipedia.org/wiki/Servicios_del_ecosistema) (Ecosystem Services). Peso de 10%.
- Pesca ([Fisheries](https://en.wikipedia.org/wiki/Fishery)). Peso de 10%.
- [Cambio climatico](https://es.wikipedia.org/wiki/Cambio_clim%C3%A1tico) (Climate Change). Peso de 40%.
- Emisiones contaminantes ([Pollution Emissions](https://en.wikipedia.org/wiki/Air_pollution)). Peso de 5%.
- [Agricultura](https://es.wikipedia.org/wiki/Agricultura) (Agriculture). Peso de 5%.
- [Recursos hídricos](https://es.wikipedia.org/wiki/Recurso_h%C3%ADdrico). (Water Resources). Peso de 5%.

-----------------

Por lo tanto para calcular el EPI se utiliza una [media ponderada](https://es.wikipedia.org/wiki/Media_ponderada) de los 2 indicadores principales. A su vez se calcula una media ponderada con los indicadores internos de ambos. 

In [11]:
df_epi_region[['country', 'region', 'EPI.rgn.rank', 'EPI.rgn.mean', 'EPI.new', 'HLT.new', 'ECO.new']]

country                     region  ...  HLT.new  ECO.new
0             Afghanistan              Southern Asia  ...     20.0     29.2
1                  Angola         Sub-Saharan Africa  ...     20.4     35.9
2                 Albania             Eastern Europe  ...     44.5     52.0
3    United Arab Emirates        Greater Middle East  ...     55.2     55.9
4               Argentina  Latin America & Caribbean  ...     60.2     46.8
..                    ...                        ...  ...      ...      ...
175               Vanuatu               Asia-Pacific  ...     26.7     30.4
176                 Samoa               Asia-Pacific  ...     42.4     33.9
177          South Africa         Sub-Saharan Africa  ...     31.1     51.0
178                Zambia         Sub-Saharan Africa  ...     21.0     43.9
179              Zimbabwe         Sub-Saharan Africa  ...     22.5     46.7

[180 rows x 7 columns]

In [12]:
df_epi_group = pd.read_csv('epi2020resultsgroups20200604.csv')
df_epi_group.head(5)

code  iso  ... WWT.sids.rank WWT.lldc.rank
0     4  AFG  ...           NaN          17.0
1    24  AGO  ...           NaN           NaN
2     8  ALB  ...           NaN           NaN
3   784  ARE  ...           NaN           NaN
4    32  ARG  ...           NaN           NaN

[5 rows x 1352 columns]

Aparentemente se ven columnas adicionales, utilizadas para obtener diferentes indices. No lo tomares a este dataframe.

In [13]:
df_epi_attr = pd.read_csv('epi2020countryattributes20200604.csv')
df_epi_attr.head(5) 

code  iso      country               region  ...  oei  oic  opec  emmrkt
0     4  AFG  Afghanistan        Southern Asia  ...    0    1     0       0
1     8  ALB      Albania       Eastern Europe  ...    0    1     0       0
2    12  DZA      Algeria  Greater Middle East  ...    0    1     1       0
3    20  AND      Andorra          Global West  ...    1    0     0       0
4    24  AGO       Angola   Sub-Saharan Africa  ...    0    0     1       0

[5 rows x 20 columns]

In [14]:
df_epi_attr.shape

(220, 20)

Se puede observar que este dataset se encuentra dentro del dataset **epi2020resultsgroups**, asique no lo utilizaremos.

In [15]:
df_epi_weights = pd.read_csv('epi2020weights20200604.csv')
df_epi_weights.head(10)

Type Abbreviation  ... MaterialityFilter  Order
0              EPI          EPI  ...               NaN      1
1  PolicyObjective          HLT  ...               NaN      2
2    IssueCategory          AIR  ...               NaN      3
3        Indicator          PMD  ...               NaN      4
4        Indicator          HAD  ...               NaN      5
5        Indicator          OZD  ...               NaN      6
6    IssueCategory          H2O  ...               NaN      7
7        Indicator          USD  ...               NaN      8
8        Indicator          UWD  ...               NaN      9
9    IssueCategory          HMT  ...               NaN     10

[10 rows x 7 columns]

In [16]:
df_epi_weights.shape

(46, 7)

## Conociendo Human Freedom Index (HFI)


In [17]:
!ls '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom/data-2020'

data-america-2020.csv	    data-north-africa-middle-east-2020.csv
data-asia-pacific-2020.csv  data-sub-saharan-africa-2020.csv
data-europe-2020.csv


In [18]:
!ls '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom/'

data-2020	  data-asia-pacific.csv  data-north-africa-middle-east.csv
data-america.csv  data-europe.csv	 data-sub-saharan-africa.csv


Ahora estudiemos un dataset de todos los que hay para saber el formato de los mismos. 

In [19]:
cd '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom/'

/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom


In [20]:
df_america = pd.read_csv('data-america.csv')
df_america.head(3)

Name  Index Year  ...  Investment Freedom  Financial Freedom
0  Argentina        2020  ...                55.0               60.0
1   Barbados        2020  ...                70.0               60.0
2     Belize        2020  ...                55.0               50.0

[3 rows x 15 columns]

In [21]:
df_america.shape

(832, 15)

Podemos observar que el dataframe tiene 832 filas y 15 columnas. Veamos las columnas:

In [22]:
df_america.columns

Index(['Name', 'Index Year', 'Overall Score', 'Property Rights',
       'Judicial Effectiveness', 'Government Integrity', 'Tax Burden',
       'Government Spending', 'Fiscal Health', 'Business Freedom',
       'Labor Freedom', 'Monetary Freedom', 'Trade Freedom',
       'Investment Freedom', 'Financial Freedom'],
      dtype='object')

Estudiaremos los indicadores utilizados en el indice de Libertad Economica. [Indice de Libertad Economica - Español](https://es.wikipedia.org/wiki/%C3%8Dndice_de_Libertad_Econ%C3%B3mica), [Index of Economic Freedom- English](https://en.wikipedia.org/wiki/Index_of_Economic_Freedom#Investment_freedom)

¿Que significa cada columna?

En el siguiente informe, [2020 Index of Economic Freedom HIGHLIGHTS](https://www.heritage.org/index/pdf/2020/book/2020_IndexofEconomicFreedom_Highlights.pdf), se divide en 4 partes las 12 columnas desde Proporty Rights hasta Financial Freedom. A estas 12 columnas se las llama, **THE 12 ECONOMIC FREEDOMS** o **LAS 12 LIBERTADES ECONOMICAS**. Estudiemos un poco cada una de ellas. No se agregará información al documento, habrán enlaces que nos redirigen a información oficial. 

----------

- [Rule of law](https://www.heritage.org/index/rule-of-law) - [Imperio de la ley](https://es.wikipedia.org/wiki/Imperio_de_la_ley)

1. [Property Rights](https://www.heritage.org/index/property-rights) - Derechos de propiedad
2. Judicial Effectiveness - Efectividad judicial
3. [Government Integrity](https://www.heritage.org/index/freedom-from-corruption) - Integridad del gobierno

----------

- [Government Size](https://www.heritage.org/index/limited-government) - Tamaño del [Gobierno](https://es.wikipedia.org/wiki/Gobierno) ([Big Government](https://en.wikipedia.org/wiki/Big_government) or [Small Government](https://en.wikipedia.org/wiki/Small_government))

4. [Tax Burden](https://www.heritage.org/index/fiscal-freedom) - Carga fiscal
5. [Government Spending](https://www.heritage.org/index/government-spending) - Gastos gubernamentales 
6. Fiscal Health - Salud Fiscal

----------

- [Regulatory Efficiency](https://www.heritage.org/index/regulatory-efficiency) - Eficiencia Regulatoria

7. [Business Freedom](https://www.heritage.org/index/business-freedom) - Libertad comercial
8. [Labor Freedom](https://www.heritage.org/index/labor-freedom) - Libertad labotal
9. [Monetary Freedom](https://www.heritage.org/index/monetary-freedom) - Libertad monetaria

----------

- [Open Markets](https://www.heritage.org/index/open-markets) - Apertura de mercados

10. [Trade Freedom](https://www.heritage.org/index/trade-freedom.aspx) - [Libertad comercial](https://es.wikipedia.org/wiki/Libre_comercio)
11. [Investment Freedom](https://www.heritage.org/index/investment-freedom) - Libertad de inversión.
12. [Financial Freedom](https://www.heritage.org/index/financial-freedom) - Libertad financiera

----------

Veamos algo de información del dataframe

In [23]:
df_america.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 832 entries, 0 to 831
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Name                    832 non-null    object 
 1   Index Year              832 non-null    int64  
 2   Overall Score           787 non-null    float64
 3   Property Rights         787 non-null    float64
 4   Judicial Effectiveness  128 non-null    float64
 5   Government Integrity    787 non-null    float64
 6   Tax Burden              787 non-null    float64
 7   Government Spending     787 non-null    float64
 8   Fiscal Health           128 non-null    float64
 9   Business Freedom        787 non-null    float64
 10  Labor Freedom           500 non-null    float64
 11  Monetary Freedom        787 non-null    float64
 12  Trade Freedom           787 non-null    float64
 13  Investment Freedom      787 non-null    float64
 14  Financial Freedom       787 non-null    fl

Agregamos una nueva columna para identificar la region de este dataframe. La misma nos ayudará a futuros analisis por regiones. 

In [24]:
df_america['Region'] = 'America'
df_america

Name  Index Year  ...  Financial Freedom   Region
0              Argentina        2020  ...               60.0  America
1               Barbados        2020  ...               60.0  America
2                 Belize        2020  ...               50.0  America
3                Bolivia        2020  ...               40.0  America
4                 Brazil        2020  ...               50.0  America
..                   ...         ...  ...                ...      ...
827          The Bahamas        1995  ...               70.0  America
828  Trinidad and Tobago        1995  ...                NaN  America
829        United States        1995  ...               70.0  America
830             Uruguay         1995  ...               50.0  America
831           Venezuela         1995  ...               70.0  America

[832 rows x 16 columns]

In [25]:
df_america['Index Year'] = pd.Categorical(df_america['Index Year'].apply(str))

### Generalizando y uniendo dataframes.

Sabiendo cual es la estructura de uno de los dataset podemos generalizar a todos los demas, ya que fueron descargados de la misma fuente y de igual forma. Lo que haremos es agregar la region a cada dataframe, unirlos y luego eliminar las columnas que no se utilizarán para este estudio. Las columnas que se eliminarán son los indicadores utilizados para obtener el indice. 

> Nota: El indice HFI es una media de estos indicadores.


Crearemos una función para obtener el nombre de los archivos de las carpetas donde se encuentran contenidos los datasets.

In [26]:
from os import scandir, getcwd

def ls(ruta): # getcwd()
  '''Funcion que retorna una lista con el nombre de los archivos de la ruta que le pasemos
  Param:
    ruta : str, string de la ruta del directorio. getcwd(): para la ruta actual
  '''
  return [arch.name for arch in scandir(ruta) if arch.is_file()]

In [27]:
path_data_2020 = '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom/data-2020'
path_data = '/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom/'

Lista con los nombres de los archivos del año 2020 y del historico de 1995 a 2020.

In [28]:
nombres_datasets_hfi_2020 = ls(path_data_2020)
nombres_datasets_hfi_2020

['data-america-2020.csv',
 'data-asia-pacific-2020.csv',
 'data-north-africa-middle-east-2020.csv',
 'data-europe-2020.csv',
 'data-sub-saharan-africa-2020.csv']

In [29]:
nombres_datasets_hfi = ls(path_data)
nombres_datasets_hfi

['data-europe.csv',
 'data-america.csv',
 'data-sub-saharan-africa.csv',
 'data-north-africa-middle-east.csv',
 'data-asia-pacific.csv']

#### Año 2020

Justamos todos los dataframes del año 2020 en uno, tambien le vamos a agregar una columna nueva llamada *Region*. El nombre de la region lo obtenemos del titulo de los csv's. 

In [30]:
import os 

In [31]:
os.chdir(path_data_2020)

In [32]:
os.getcwd()

'/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom/data-2020'

In [33]:
import re

pattern = 'data-([\w\W]+)-2020.csv'

lista_de_datasets = []
regiones = []

for data_set in nombres_datasets_hfi_2020:
  region = re.findall(pattern, data_set)
  df = pd.read_csv(data_set)
  df['Region'] = region[0]
  df['Index Year'] = pd.Categorical(df['Index Year'].apply(str))
  lista_de_datasets.append(df)

df_hfi_2020 = pd.concat(lista_de_datasets)
columnas = range(3, len(df_hfi_2020.columns) - 1)
df_hfi_2020 = df_hfi_2020.drop(df_hfi_2020.columns[columnas], axis=1)
df_hfi_2020.head(3)


Name Index Year  Overall Score   Region
0  Argentina       2020           53.1  america
1   Barbados       2020           61.4  america
2     Belize       2020           57.4  america

#### Historico 1995-2020

Justamos todos los dataframes desde 1995 al año 2020 en uno, tambien le vamos a agregar una columna nueva llamada *Region*. El nombre de la region lo obtenemos del titulo de los csv's.

In [34]:
os.chdir(path_data)

In [35]:
os.getcwd()

'/content/drive/My Drive/Colab Notebooks/republica-liberal/data/index_economic_freedom'

In [36]:
import re

pattern = 'data-([\w\W]+).csv'

lista_de_datasets = []
regiones = []

for data_set in nombres_datasets_hfi:
  region = re.findall(pattern, data_set)
  df = pd.read_csv(data_set)
  df['Region'] = region[0]
  df['Index Year'] = pd.Categorical(df['Index Year'].apply(str))
  lista_de_datasets.append(df)

df_hfi = pd.concat(lista_de_datasets)
columnas = range(3, len(df_hfi.columns) - 1)
df_hfi = df_hfi.drop(df_hfi.columns[columnas], axis=1)
df_hfi

Name Index Year  Overall Score        Region
0          Albania       2020           66.9        europe
1          Armenia       2020           70.6        europe
2          Austria       2020           73.3        europe
3       Azerbaijan       2020           69.3        europe
4          Belarus       2020           61.7        europe
...            ...        ...            ...           ...
1068         Tonga       1995            NaN  asia-pacific
1069  Turkmenistan       1995            NaN  asia-pacific
1070    Uzbekistan       1995            NaN  asia-pacific
1071       Vanuatu       1995            NaN  asia-pacific
1072       Vietnam       1995           41.7  asia-pacific

[4784 rows x 4 columns]

In [37]:
df_hfi.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4784 entries, 0 to 1072
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype   
---  ------         --------------  -----   
 0   Name           4784 non-null   object  
 1   Index Year     4784 non-null   category
 2   Overall Score  4263 non-null   float64 
 3   Region         4784 non-null   object  
dtypes: category(1), float64(1), object(2)
memory usage: 155.6+ KB


# Selección de datasets. Limpiando datasets.

Utilizaremos HFI 2020, HFI 1995-2020, EPI 1950 - 2020. 

## HFI 1995-2020

In [38]:
df_hfi.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4784 entries, 0 to 1072
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype   
---  ------         --------------  -----   
 0   Name           4784 non-null   object  
 1   Index Year     4784 non-null   category
 2   Overall Score  4263 non-null   float64 
 3   Region         4784 non-null   object  
dtypes: category(1), float64(1), object(2)
memory usage: 155.6+ KB


## HFI 2020

In [39]:
df_hfi_2020.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 186 entries, 0 to 47
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype   
---  ------         --------------  -----   
 0   Name           186 non-null    object  
 1   Index Year     186 non-null    category
 2   Overall Score  180 non-null    float64 
 3   Region         186 non-null    object  
dtypes: category(1), float64(1), object(2)
memory usage: 6.1+ KB


## EPI por Region

Utilizaremos el dataframe epi, con la columna adicional de las regiones.

In [40]:
df_epi_region.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Columns: 142 entries, code to WWT.rgn.rank
dtypes: float64(92), int64(47), object(3)
memory usage: 199.8+ KB


In [42]:
df_epi_clean = df_epi_region.drop(['code', 'iso'], axis = 1)

Borraremos todas las columnas y solo dejaremos las del indice EPI y sus subindices ECO y HLT

In [44]:
list(df_epi_clean.columns)

['country',
 'region',
 'EPI.new',
 'HLT.new',
 'AIR.new',
 'PMD.new',
 'HAD.new',
 'OZD.new',
 'H2O.new',
 'USD.new',
 'UWD.new',
 'HMT.new',
 'PBD.new',
 'WMG.new',
 'MSW.new',
 'ECO.new',
 'BDH.new',
 'TBN.new',
 'TBG.new',
 'MPA.new',
 'PAR.new',
 'SHI.new',
 'SPI.new',
 'BHV.new',
 'ECS.new',
 'TCL.new',
 'GRL.new',
 'WTL.new',
 'FSH.new',
 'FSS.new',
 'RMS.new',
 'FGT.new',
 'CCH.new',
 'CDA.new',
 'CHA.new',
 'FGA.new',
 'NDA.new',
 'BCA.new',
 'LCB.new',
 'GIB.new',
 'GHP.new',
 'APE.new',
 'SDA.new',
 'NXA.new',
 'AGR.new',
 'SNM.new',
 'WRS.new',
 'WWT.new',
 'EPI.rgn.mean',
 'EPI.rgn.rank',
 'HLT.rgn.mean',
 'HLT.rgn.rank',
 'AIR.rgn.mean',
 'AIR.rgn.rank',
 'PMD.rgn.mean',
 'PMD.rgn.rank',
 'HAD.rgn.mean',
 'HAD.rgn.rank',
 'OZD.rgn.mean',
 'OZD.rgn.rank',
 'H2O.rgn.mean',
 'H2O.rgn.rank',
 'USD.rgn.mean',
 'USD.rgn.rank',
 'UWD.rgn.mean',
 'UWD.rgn.rank',
 'HMT.rgn.mean',
 'HMT.rgn.rank',
 'PBD.rgn.mean',
 'PBD.rgn.rank',
 'WMG.rgn.mean',
 'WMG.rgn.rank',
 'MSW.rgn.mean',


In [45]:
columns_selected = ['country','region',
                    'EPI.new', 'EPI.rgn.mean', 'EPI.rgn.rank',
                    'ECO.new', 'ECO.rgn.mean', 'ECO.rgn.rank',
                    'HLT.new', 'HLT.rgn.mean', 'HLT.rgn.rank']

In [57]:
df_epi_clean = df_epi_clean[columns_selected]
df_epi_clean.head(5)

country                     region  ...  HLT.rgn.mean  HLT.rgn.rank
0           Afghanistan              Southern Asia  ...     26.775000             6
1                Angola         Sub-Saharan Africa  ...     22.734783            24
2               Albania             Eastern Europe  ...     58.426316            17
3  United Arab Emirates        Greater Middle East  ...     48.737500             5
4             Argentina  Latin America & Caribbean  ...     46.843750             5

[5 rows x 11 columns]

In [58]:
df_epi_clean['region'].unique()

array(['Southern Asia', 'Sub-Saharan Africa', 'Eastern Europe',
       'Greater Middle East', 'Latin America & Caribbean',
       'Former Soviet States', 'Global West', 'Asia-Pacific'],
      dtype=object)

Apuntes:

reportChapter 3. Explaining Performance - Pagina 39 de 

https://epi.yale.edu/downloads/epi2020report20200828.pdf


Explicando la performance. Se debe principalmente a los paises mas ricos. 

>  this correlation may not be surprising, given that improved outcomes on many dimen-sions of sustainability require invest-ments in infrastructure, administrative capacity, and human capital, which wealthier countries are better posi-tioned to make. 